ci/main.go

```diff
@@ -0,0 +1,211 @@
+	// if err != nil {
+	// 	return (err)
+	// }
+	// fmt.Println(contents)
+
+	img, err = buildProtocGenParigot(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	img, err = generateRep(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	img, err = buildFilePWasm(ctx, img)
+	if err != nil {
+		return err
+	}
+
+	// get reference to build output directory in container
+	output := img.Directory("build")
+
+	// write contents of container build/ directory to the host
+	_, err = output.Export(ctx, "build")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildProtocGenParigot(ctx context.Context, img *dagger.Container) (*dagger.Container, error) {
+	/*
+		TEMPLATE=$(shell find command/protoc-gen-parigot -type f -regex ".*\.tmpl")
+		GENERATOR_SRC=$(shell find command/protoc-gen-parigot -type f -regex ".*\.go")
+		build/protoc-gen-parigot: $(TEMPLATE) $(GENERATOR_SRC)
+			@rm -f $@
+			$(GO_TO_HOST) build $(EXTRA_HOST_ARGS) -o $@ github.com/iansmith/parigot/command/protoc-gen-parigot
+	*/
+
+	var err error
+	// template, err = findFilesEndWith("command/protoc-gen-parigot", ".tmpl")
+	// if err != nil {
+	// 	return img, err
+	// }
+	// eneratorSrc, err = findFilesEndWith("test", ".go")
+	// if err != nil {
+	// 	return img, err
+	// }
+	template, err = img.WithExec([]string{"bash", "-c", `find command/protoc-gen-parigot -type f -regex ".*\.tmpl"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+	eneratorSrc, err = img.WithExec([]string{"bash", "-c", `find command/protoc-gen-parigot -type f -regex ".*\.go"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+
+	// contents, err := img.WithExec([]string{"ls", "-l", "."}).Stdout(ctx)
+	// if err != nil {
+	// 	return img, err
+	// }
+	// fmt.Println(contents)
+
+	target := "build/protoc-gen-parigot"
+	packagePath := "github.com/iansmith/parigot/command/protoc-gen-parigot"
+	img = img.WithExec([]string{"rm", "-f", target})
+	img = img.WithExec([]string{GO2HOST, "build", "-o", target, packagePath})
+
+	return img, nil
+}
+
+func generateRep(ctx context.Context, img *dagger.Container) (*dagger.Container, error) {
+	/*
+		API_PROTO=$(shell find api/proto -type f -regex ".*\.proto")
+		TEST_PROTO=$(shell find test -type f -regex ".*\.proto")
+
+		## we just use a single representative file for all the generated code from
+		REP=g/file/$(API_VERSION)/file.pb.go
+		$(REP): $(API_PROTO) $(TEST_PROTO) build/protoc-gen-parigot
+			@rm -rf g/*
+			buf lint
+			buf generate
+	*/
+
+	var err error
+	apiProto, err = img.WithExec([]string{"bash", "-c", `find api/proto -type f -regex ".*\.proto"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+	testProto, err = img.WithExec([]string{"bash", "-c", `find test -type f -regex ".*\.proto"`}).Stdout(ctx)
+	if err != nil {
+		return img, err
+	}
+
+	img = img.WithExec([]string{"rm", "-rf", "g/*"})
+	img = img.WithExec([]string{"buf", "lint"})
+	img = img.WithExec([]string{"buf", "generate"})
```

Comment:
## JUST FYI


The lines above, just keep in mind the issue with `rm`  and searching for files.

These two commands are absolutely critical.

```
	img = img.WithExec([]string{"buf", "lint"})
 	img = img.WithExec([]string{"buf", "generate"})
```

Those end up invoking protoc-gen-parigot which is a compiler that outputs a bunch of the boilerplate to keep users from having to see about 90% of how parigot works.